lib/cocoapods/installer.rb

```diff
@@ -212,7 +212,9 @@ def generate_pods_project(generator = create_generator)
     #         generated as result of the analyzer.
     #
     def pod_targets
-      aggregate_targets.map(&:pod_targets).flatten.uniq
+      aggregate_target_pod_targets = aggregate_targets.map(&:pod_targets).flatten
```

Comment:
the `installer` always took for granted that the set of aggregate targets include the full set of pod targets. This is no longer true because we can have pod targets that are only used for tests and therefore should not be included in the aggregate targets. 

Added a test for this and also checked existing tests still work here.